projects/scion/microfrontend-platform/src/lib/client/messaging/message-handler.ts

```diff
@@ -103,3 +101,11 @@ export class MessageHandler<MSG extends Message, REPLY> {
       }));
   }
 }
+
+function coerceObservable<T>(value: T | Observable<T> | Promise<T>): Observable<T> {
```

Comment:
Wouldn't it be better if we explicitly listed the cases?

```ts
/**
 * Creates an Observable from the result of the callback.
 *
 * If the callback returns no value (`void`), returns `undefined`, or returns a Promise that resolves to `undefined`,
 * the returned observable will complete immediately.
 */
function fromCallbackResult$<T>(value: T | Observable<T> | Promise<T>): Observable<T> {
  if (value === undefined) {
    return EMPTY;
  }
  if (value instanceof Observable) {
    return value;
  }
  if (value instanceof Promise) {
    return from(value).pipe(filter(resolved => resolved !== undefined));
  }
  return of(value);
}
```

Reply:
What do you think of the following commit message?

```
fix(platform): mirror Observable if replying with an Observable in message or intent callback

Previously, the communication was terminated when the Observable emitted `undefined`.
```

Reply:
done